tests/shared/src/test/scala/org/http4s/PathSuite.scala

```diff
@@ -28,16 +28,24 @@ class PathSuite extends Http4sSuite {
   test("equals should be consistent with equality") {
     forAll { (a: Path, b: Path) =>
       if (a == b)
-        (a.segments == b.segments) && (a.absolute == b.absolute) && (a.endsWithSlash == b.endsWithSlash)
+        assert(
+          (a.segments == b.segments) && (a.absolute == b.absolute) && (a.endsWithSlash == b.endsWithSlash)
+        )
       else
-        (a.segments != b.segments) || (a.absolute != b.absolute) || (a.endsWithSlash != b.endsWithSlash)
+        assert(
+          (a.segments != b.segments) || (a.absolute != b.absolute) || (a.endsWithSlash != b.endsWithSlash)
+        )
     }
   }
 
   test("hashcode should be consistent with equality") {
     forAll { (a: Path, b: Path) =>
-      if (a == b) a.## == b.##
-      else a.## != b.##
+      assert(
+        if (a.## != b.##)
+          a != b // if the hashCodes are different, then the Paths should be different by equality
+        else
+          (a == b) && (a.## == b.##) // if the Paths are equal, then the hashcodes should be the same
+      )
```

Comment:
I wonder if this law is expressing the same idea, the expression is so concise I'm not exactly sure 😂 in any case, I think it would be good to run the `HashTests` for `Path`.
https://github.com/typelevel/cats/blob/771c6c802f59c72dbc1be1898081c9c882ddfeb0/kernel-laws/shared/src/main/scala/cats/kernel/laws/HashLaws.scala#L30-L31

Reply:
And then, that's where you learn about something you didn't know existed :-D
Well, yes, that looks like it does!
While I'm on that, I'm thinking that maybe also the test above (`equals should be consistent with equality`) could be evicted by running `EqTests` for `Path`? https://github.com/typelevel/cats/blob/771c6c802f59c72dbc1be1898081c9c882ddfeb0/kernel-laws/shared/src/main/scala/cats/kernel/laws/EqLaws.scala
wdyt?

Reply:
Ah, yes, sounds good! Delegating to law tests is always good.